internal/saucecloud/xcuitest.go

```diff
@@ -106,3 +116,35 @@ func (r *XcuitestRunner) calculateJobsCount(suites []xcuitest.Suite) int {
 	}
 	return jobsCount
 }
+
+// ensureAppsAreIpa checks if apps are a .ipa package. Otherwise, it generates one.
+func ensureAppsAreIpa(appPath, testAppPath *string) error {
```

Comment:
The description doesn't mention that `appPath` and `testAppPath` are potentially modified. I know that the pointer enables and therefore suggests that possibility, but it'd be good to state this behavior explicitly, as it's unclear _when_ and for _what_ purpose those strings are modified without reading the actual code. Especially important since the modification doesn't even occur in this method, but nested in another call further down the chain.

Reply:
@alexplischke I've reorganised the code, so that will be really easier to understand, without any variable overwriting.